test/setup/main.tf

```diff
@@ -30,8 +30,35 @@ module "project_ci_kms" {
     "compute.googleapis.com",
     "iam.googleapis.com",
     "artifactregistry.googleapis.com",
-    "cloudbuild.googleapis.com",
     "cloudresourcemanager.googleapis.com",
     "cloudbilling.googleapis.com"
   ]
+
+  activate_api_identities = [
+    {
+      api   = "cloudbuild.googleapis.com",
+      roles = ["roles/cloudbuild.builds.builder"]
+    }
+  ]
+}
+
+data "google_client_config" "default" {}
+
+resource "terracurl_request" "poke" {
```

Comment:
Possibly silly question, what is this Cloud Build POST curl request for? Let's add a comment for future reference, if you have a doc link or quick explanation that'd be great!

Reply:
This is a good question! Just added the comment. Basically, this resource solved the issue I was facing before regarding the cloud build service agent error (not exists when I tried to use).

I figure it out that when we enable Cloud Build API, the CB service account (`[PROJECT_NUMBER]@cloudbuild.gserviceaccount.com`) is created, but not the service agent (`service-PROJECT_NUMBER@gcp-sa-cloudbuild.iam.gserviceaccount.com`). Looks like CB service agent is a "Just-in-time" (JIT) provisioning resource.

Reply:
Aah, got it! That's good to know, thanks for the context 👍